documentation/src/docs/controller-service.md

```diff
@@ -282,168 +277,193 @@ Controller service efficient queries for finding immediate successors
+
+**Note**: To retrieve Stream Segment record given a Stream Segment ID, we first need to extract the creation epoch and then retrieve the Stream Segment record from the epoch record.
 
 #### Stream Configuration
- Znode under which stream configuration is serialized and persisted. A
- stream configuration contains stream policies that need to be
+ Znode under which Stream configuration is serialized and persisted. A
+ Stream configuration contains Stream policies that need to be
  enforced.
  Scaling policy and Retention policy are supplied by the application at
- the time of stream creation and enforced by Controller by monitoring
- the rate and size of data in the stream.
+ the time of Stream creation and enforced by Controller by monitoring
+ the rate and size of data in the Stream.
 
    - Scaling policy describes if and when to automatically scale based on
- incoming traffic conditions into the stream. The policy supports two
- flavours - _traffic as rate of events per second_ and _traffic as rate of
+ incoming traffic conditions into the Stream. The policy supports two
+ flavours - _traffic as rate of Events per second_ and _traffic as rate of
  bytes per second_. The application specifies their desired traffic
  rates into each segment by means of scaling policy and the supplied
  value is chosen to compute thresholds that determine when to scale a
- given stream.
+ given Stream.
 
    - Retention Policy describes the amount of data that needs to be
- retained into Pravega cluster for this stream. We support a time based
- and a size based retention policy where applications can choose
- whether they want to retain data in the stream by size or by time by
+ retained into Pravega cluster for this Stream. We support a _time based_
+ and a _size based_ retention policy where applications can choose
+ whether they want to retain data in the Stream by size or by time by
  choosing the appropriate policy and supplying their desired values.
 
 #### Stream State
- Znode which captures the state of the stream. It is an enumerator with
+ Znode which captures the state of the Stream. It is an enumerator with
  values from *creating, active, updating, scaling, truncating, sealing,*
- and *sealed*. Once _active_, a stream transitions between performing a
- specific operation and active until it is sealed. A transition map is
+ and *sealed*. Once _active_, a Stream transitions between performing a
+ specific operation and remanis _active_ until it is sealed. A transition map is
  defined in the
- [State](https://github.com/pravega/pravega/blob/master/Controller/src/main/java/io/pravega/controller/store/stream/tables/State.java)
+ [State](https://github.com/pravega/pravega/blob/master/Controller/src/main/java/io/pravega/controller/store/stream/State.java)
```

Comment:
@shiveshr Hyperlink throws error...link not found

Reply:
https://github.com/pravega/pravega/blob/master/controller/src/main/java/io/pravega/controller/store/stream/State.java

Reply:
Need to check